### DISCUSSION TITLE
Bills, Recurring transactions and Rules

### DISCUSSION THREAD

**Discussion by PAS-BC** [firefly-iii/firefly-iii#9151] [Firefly III]
I have a Bill created, with a Recurring Transaction to suit.
There is also a Rule associated with the Bill.

I am only just getting familiar with FF so my problem is probably just lack of understanding.

A Bill was due yesterday, but the expected transaction did not appear in the relevant Account.

Furthermore, the 'Bills' list now shows "Expected 1 day ago" in the 'Paid this period' column.

I have refreshed Cache, done several 'docker compose pull' & 'up -d' processes to 'jolt' FF into action - but no luck.

Suppose I will just have to do a manual 'Withdrawal' to record the payment for the Bill, but curious as to what went wrong - and hope it works next month.

Grateful for any tips.

**Comment by JC5** [comment]
Are you running the cron job?

https://docs.firefly-iii.org/how-to/firefly-iii/advanced/cron/

**Comment by PAS-BC** [comment]
Good morning James,

Thank you for the quick reply.

 

As far as I know, yes.

I have two recurring deposits which are working ok.

**Comment by PAS-BC** [comment]
I think this shows it is running (from ‘docker container ls’)?

 

149c871f8a0b   alpine                                               "sh -c 'echo \"0 3 * …"   3 days ago       Up 3 days                                                                                                                                                                   firefly_iii_cron

**Comment by PAS-BC** [comment]
Output from docker logs on the Cron job:

```
crond: crond (busybox 1.36.1) started, log level 8
crond: USER root pid   8 cmd wget -qO- http://fireflyiii:8080/api/v1/cron/secretcode
{"recurring_transactions":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Recurring transactions cron job fired successfully."},"auto_budgets":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Auto-budget cron job fired successfully."},"bill_warnings":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Bill warning cron job fired successfully."}}crond: USER root pid   9 cmd wget -qO- http://fireflyiii:8080/api/v1/cron/secretcode
{"recurring_transactions":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Recurring transactions cron job fired successfully."},"auto_budgets":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Auto-budget cron job fired successfully."},"bill_warnings":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Bill warning cron job fired successfully."}}crond: USER root pid  10 cmd wget -qO- http://fireflyiii:8080/api/v1/cron/secretcode
```

**Comment by JC5** [comment]
Sorry, no clue. You say you have a recurring transaction, I'm assuming it was supposed to be created yesterday, but it didn't, is that it?

**Comment by PAS-BC** [comment]
Yep, that's the problem.

I watched it a lot yesterday (18th August) as I was doing my first big reconciliation.

It kept showing 'next due 18th August'.

I looked again after midnight and it showed 'due a day ago' - hence my 'WTF' moment 😁

⁣Get BlueMail for Android ​

**Reply by JC5** [reply]
OK, so what are the logs? Was there an error? Is you recurring transaction configured correctly?

**Comment by PAS-BC** [comment]
Looking at the various FF logs via Portainer, there are no signs of errors.
However I have done a couple of 'docker compose' commands while trying to 'trigger' the transaction in question, so I guess those logs are meaningless.

When I look at the Recurring Transaction Overview for the item in question, the 'next expected' is for next month. So it figures that 'this month' should have fired. Indeed the list has that 'Expected 1 day ago' status, like "well, it was to supposed to happen..."

The settings of the transaction in question follow the same format as a couple of recurring transactions that work for deposits.
The deposits are fortnightly, so there is already some history of them working.
All the other recurring transactions are monthly or yearly, so they don't yet have a track record.

I could just do a manual entry for the monthly item that got 'stuck', and see what happens next month.
It will hopefully clear the 'Expected 1 day ago' flag anyway.

Are there any other FF logs I should generate?
I skimmed over 'logs' generally in the .env file prior to install.

**Comment by PAS-BC** [comment]
I did the manual entry and will keep an eye on it next month.
